lib/utils.pm

```diff
@@ -421,6 +422,27 @@ sub is_bridged_networking {
     return $ret;
 }
 
+=head2 set_hostname
+    set_hostname($hostname);
+
+Setting hostname according input parameter using hostnamectl.
+Calling 'reload-or-restart' to make sure that network stack will propogate
+hostname into DHCP/DNS
+
+if you change hostname using `hostnamectl set-hostname`, then `hostname -f` will fail with "hostname: Name or service not known"
+also DHCP/DNS don't know about the changed hostname, you need to send a new DHCP request to update dynamic DNS
+yast2-network module does "NetworkService.ReloadOrRestart if Stage.normal || !Linuxrc.usessh" if hostname is changed via `yast2 lan`
+=cut
+sub set_hostname {
+    my ($hostname) = @_;
+    assert_script_run "hostnamectl set-hostname $hostname";
+    assert_script_run "hostnamectl status|grep $hostname";
+    assert_script_run "hostname|grep $hostname";
+    systemctl 'status network.service';
+    save_screenshot;
+    assert_script_run "if systemctl -q is-active network.service; then systemctl reload-or-restart network.service; fi";
```

Comment:
how about just `systemctl -q is-active network.service && systemctl reload-or-restart network.service || true` ?

Reply:
you are welcome to do another pull request and test this , I just refactor this code into separate function . Meaning I cut it in one place and move to more suitable